### PR TITLE
Tweak schema validation code for readability.

### PR DESCRIPTION
**Pull request by davinchia**
## What
I was reading this piece of code on the plane while prepping for a presentation and felt some slight tweaks would really help with readability.

## How
See inline comments.

## Recommended reading order
The one file.

### CODE REVIEW

File: airbyte-workers/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java [hunk 1]
```diff
@@ -386,6 +366,36 @@ private static Runnable getReplicationRunnable(final AirbyteSource source,
     };
   }
 
+  private static void validateSchema(RecordSchemaValidator recordSchemaValidator,
+                                     Map<String, ImmutablePair<Set<String>, Integer>> validationErrors,
+                                     AirbyteMessage message) {
+    if (message.getRecord() == null) {
```
Comment (davinchia) [inline airbyte-workers/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java:372]: exit early to reduce nested braces.

File: airbyte-workers/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java [hunk 2]
```diff
@@ -386,6 +366,36 @@ private static Runnable getReplicationRunnable(final AirbyteSource source,
...
+      return;
+    }
+
+    final AirbyteRecordMessage record = message.getRecord();
+    final String messageStream = WorkerUtils.streamNameWithNamespace(record.getNamespace(), record.getStream());
+    // avoid noise by validating only if the stream has less than 10 records with validation errors
+    final boolean streamHasLessThenTenErrs =
+        validationErrors.get(messageStream) == null || validationErrors.get(messageStream).getRight() < 10;
```
Comment (davinchia) [inline airbyte-workers/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java:380]: turn the condition into a bool var so the name helps with readability.

File: airbyte-workers/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java [hunk 3]
```diff
@@ -306,31 +306,11 @@ private static Runnable getReplicationRunnable(final AirbyteSource source,
...
-                  }
-                }
-
-              }
-            }
 
-            final AirbyteMessage message = mapper.mapMessage(messageOptional.get());
+          if (messageOptional.isPresent()) {
+            final AirbyteMessage airbyteMessage = messageOptional.get();
+            validateSchema(recordSchemaValidator, validationErrors, airbyteMessage);
```
Comment (davinchia) [inline airbyte-workers/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java:312]: pull all this logic into a separate function to keep logic at the same detail layers e.g. the top level replication function should only know about high-level replication related code.

**Review by lmossman — APPROVED** [review]
Definitely an improvement, thanks for the refactor!

**Pull request merged**